unblob/handlers/filesystem/yaffs/utils.py

```diff
@@ -0,0 +1,574 @@
+        self.end_offset = -1
+        if config is None:
+            self.config = self.auto_detect()
+            logger.debug("auto-detected config", config=self.config)
+        else:
+            self.config = config
+
+    def bruteforce(self) -> YAFFSConfig:
+        # let's do a trick here
+        entries = []
+        count = 0
+        config = YAFFSConfig(
+            endianness=Endian.LITTLE, page_size=-1, spare_size=-1, ecc=False
+        )
+        for page_size in VALID_PAGE_SIZES:
+            for spare_size in VALID_SPARE_SIZES:
+                entries = []
+                for i in range(0, self.eof // (page_size + spare_size)):
+                    start = (page_size + spare_size) * i
+                    entries.append(self.file[start : start + 10])
+
+                le_count = sum(
+                    [entry.startswith(PAGE_START_LITTLE_ENDIAN) for entry in entries]
+                )
+                be_count = sum(
+                    [entry.startswith(PAGE_START_BIG_ENDIAN) for entry in entries]
+                )
+
+                if le_count > count or be_count > count:
+                    config.endianness = (
+                        Endian.LITTLE if le_count > be_count else Endian.BIG
+                    )
+                    config.page_size = page_size
+                    config.spare_size = spare_size
+                    count = max([le_count, be_count])
+
+        if config.page_size == -1:
+            raise InvalidInputFormat("Can't find YAFFS config through bruteforce.")
+
+        return config
+
+    def auto_detect(self) -> YAFFSConfig:
+        """Auto-detect page_size, spare_size, and ECC using known signatures."""
+        config = None
+        page_size = 0
+        for page_size in VALID_PAGE_SIZES:
+            if self.file[page_size:].startswith(SPARE_START_LITTLE_ENDIAN_ECC):
+                config = YAFFSConfig(
+                    endianness=Endian.LITTLE,
+                    page_size=page_size,
+                    ecc=True,
+                    spare_size=-1,
+                )
+                break
+            if self.file[page_size:].startswith(SPARE_START_LITTLE_ENDIAN_NO_ECC):
+                config = YAFFSConfig(
+                    endianness=Endian.LITTLE,
+                    page_size=page_size,
+                    ecc=False,
+                    spare_size=-1,
+                )
+                break
+            if self.file[page_size:].startswith(SPARE_START_BIG_ENDIAN_ECC):
+                config = YAFFSConfig(
+                    endianness=Endian.BIG, page_size=page_size, ecc=True, spare_size=-1
+                )
+                break
+            if self.file[page_size:].startswith(SPARE_START_BIG_ENDIAN_NO_ECC):
+                config = YAFFSConfig(
+                    endianness=Endian.BIG, page_size=page_size, ecc=False, spare_size=-1
+                )
+                break
+
+        if config is None:
+            return self.bruteforce()
```

Comment:
when can bruteforce find something, when it checks the same VALID_PAGE_SIZES?

Reply:
I got rid of the bruteforce approach. It is not required to have it.